### PR TITLE
[CLC][AMDGPU] Refactor fence helper to process order semantic explicitly

### PR DESCRIPTION
**Pull request by GeorgeWeb**
This PR refactors the builtin fence helper macro for AMDGPU to take in and process the order semantic explicitly because that is the only semantic argument accepted by the amdgcn builtin.

Additionally, makes the `None` (Monotonic) order semantic which maps to C++/SYCL's `relaxed` to be a no-op instead of falling back to the previous `acq_rel` default order.

### CODE REVIEW

**Review by JackAKirk — APPROVED** [review]
LGTM

**Comment by GeorgeWeb** [comment]
The linked unified-runtime changes https://github.com/oneapi-src/unified-runtime/pull/1414 enable support for all the memory order semantics.

**Review by kbenzie — APPROVED** [review]

**Comment by kbenzie** [comment]
@intel/llvm-gatekeepers please merge

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -130,8 +130,14 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
   )
 
   fetch_adapter_source(hip
-    ${UNIFIED_RUNTIME_REPO}
-    ${UNIFIED_RUNTIME_TAG}
+    https://github.com/GeorgeWeb/unified-runtime.git
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:133]:
```suggestion
    ${UNIFIED_RUNTIME_REPO}
```

**Review by kbenzie — CHANGES_REQUESTED** [review]

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 2]
```diff
@@ -131,7 +131,13 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
...
   fetch_adapter_source(hip
     ${UNIFIED_RUNTIME_REPO}
-    ${UNIFIED_RUNTIME_TAG}
+    # commit 2c4303c25b026f7edb215accdccb1bc5ae2e237b
+    # Merge: abe85cc9 3e011c70
+    # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
+    # Date:   Thu Jun 13 10:22:34 2024 +0100
+    #     Merge pull request #1414 from GeorgeWeb/georgi/hip-fences
+    #     [HIP] Enable more ordering and scope capabilities for atomic fences
+    2c4303c25b026f7edb215accdccb1bc5ae2e237b
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:140]:
```suggestion
    # commit 0e32bb59e0846cbe8153c294da6d8f1cfa88075b
    # Merge: 2c4303c2 08b19b22
    # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
    # Date:   Thu Jun 13 14:15:42 2024 +0100
    #     Merge pull request #1415 from GeorgeWeb/georgi/hip-atomic-memory-caps
    #     [HIP] Enable more ordering and scope capabilities for atomic memory ops
    0e32bb59e0846cbe8153c294da6d8f1cfa88075b
```

**Review by kbenzie — CHANGES_REQUESTED** [review]

**Review by kbenzie — APPROVED** [review]

**Pull request merged**